TheWarWithin/WarlockAffliction.lua

```diff
@@ -30,7 +30,21 @@ local IsSpellKnownOrOverridesKnown = C_SpellBook.IsSpellInSpellBook
 -- Specialization-specific local functions (if any)
 local Glyphed = IsSpellKnownOrOverridesKnown
 
-spec:RegisterResource( Enum.PowerType.SoulShards, {},
+spec:RegisterResource( Enum.PowerType.SoulShards, {
+        rampaging_demonic_soul = {
+        resource = "soul_shards",
+        aura = "rampaging_demonic_soul",
+
+        last = function()
+            local app = state.set_bonus.tww3 >= 4 and state.buff.rampaging_demonic_soul.applied or 0
```

Comment:
This set bonus check should be using the `set_bonus` flag (like `aura` above):

```Lua
set_bonus = "tww3_4pc",
```

(That requires confirming that the `tww3_#pc` syntax still works -- and if it doesn't, it needs to be fixed.)

Also, looks like `rampaging_demonic_soul` needs to be de-indented.

Reply:
See:  https://github.com/Hekili/hekili/blob/4fc98d684c77960e1bcd08b4e31ecf16137dbfd7/State.lua#L1450